com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -193,33 +191,43 @@ internal struct AgentParameters
         /// </summary>
         internal VectorSensor collectObservationsSensor;
 
-        /// MonoBehaviour function that is called when the attached GameObject
-        /// becomes enabled or active.
         void OnEnable()
         {
             LazyInitialize();
         }
 
+        /// <summary>
+        /// <inheritdoc cref="OnBeforeSerialize"/>
+        /// </summary>
         public void OnBeforeSerialize()
         {
+            // Manages a serialization upgrade issue from v0.13 to v0.14 where maxStep moved
+            // from AgentParameters (since removed) to Agent
             if (maxStep == 0 && maxStep != agentParameters.maxStep && !hasUpgradedFromAgentParameters)
             {
                 maxStep = agentParameters.maxStep;
             }
             hasUpgradedFromAgentParameters = true;
         }
 
+        /// <summary>
+        /// <inheritdoc cref="OnAfterDeserialize"/>
+        /// </summary>
         public void OnAfterDeserialize()
         {
+            // Manages a serialization upgrade issue from v0.13 to v0.14 where maxStep moved
+            // from AgentParameters (since removed) to Agent
             if (maxStep == 0 && maxStep != agentParameters.maxStep && !hasUpgradedFromAgentParameters)
             {
                 maxStep = agentParameters.maxStep;
             }
             hasUpgradedFromAgentParameters = true;
         }
 
+        /// <summary>
         /// Helper method for the <see cref="OnEnable"/> event, created to
```

Comment:
OnEnable is private, not sure we can have private methods as cref

Reply:
Should LazyInitialize be private too? cc: @chriselion / @surfnerd

Reply:
No, I think it should be public.

Reply:
Fixed the comment reference. Not clear why LazyInitialize needs to be public, but that discussion can be separated from this PR.